src/NuGet.Core/NuGet.Commands/PackCommandRunner.cs

```diff
@@ -186,7 +186,11 @@ private PackageArchiveReader BuildPackage(PackageBuilder builder, string outputP
 
             if (!string.IsNullOrEmpty(_packArgs.Suffix))
             {
-                builder.Version = new NuGetVersion(builder.Version.Version, _packArgs.Suffix);
+                builder.Version = new NuGetVersion($"{builder.Version.ToNormalizedString()}-{_packArgs.Suffix}");
```

Comment:
:+1: that looks like it could happen. Take a look at using the VersionFormatter to build this, V will give you the normalized version number. https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Versioning/VersionFormatter.cs#L126